### PR TITLE
Add Makefile and update README for easy installation.

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/78/
Test PASSed.

**Comment by bac** [comment]
Until we get the PPAs updated and the xenial SRU, our users will need to install the juju/charmstore-client code and build from scratch.  Putting a standard makefile in place reduces the number of steps and makes it less error-prone.

QA:

Create a novel GOPATH separate from your normal one.  Follow the steps in the README. After `go get` you'll need to update to use the version from this PR before proceeding.  When finished `$GOPATH/bin/charm` should be there.

**Comment by jrwren** [comment]
👍 QA works as described.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Makefile for the charm store client.
+
+ifndef GOPATH
```
Comment (jcsackett) [inline Makefile:3]: GOPATH, not $GOPATH?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/79/
Test PASSed.

File: Makefile (hunk 1 above)
Comment (bac) [inline Makefile:3]: Yep. Makefile....

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+$(warning You need to set up a GOPATH.)
+endif
+
+PROJECT := github.com/juju/charmstore-client
+PROJECT_DIR := $(shell go list -e -f '{{.Dir}}' $(PROJECT))
```
Comment (jcsackett) [inline Makefile:8]: This is a little opaque to me; I'm assuming it produces the path if it exists, and if not something that fails the match against CURDIR below?

File: Makefile (hunk 1 above)
Comment (jcsackett) [inline Makefile:3]: I'm not super familiar with ifndef and how it handles vars vs everywhere else. Seems there's as many different ways as there are tools that interact.

File: Makefile (hunk 2 above)
Comment (bac) [inline Makefile:8]: It is opaque but it basically is used to evaluate if go can find the project.

If it helps any, this is very well-worn trickery, cargo culted across most of our go projects.

**Comment by jcsackett** [comment]
Aside from my questions (which are just making sure I'm understanding the code right) this looks good.

:+1:

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**